app/main/views/direct_award.py

```diff
@@ -0,0 +1,193 @@
+from flask import jsonify, abort, current_app, request
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy import desc
+
+from dmapiclient.audit import AuditTypes
+from .. import main
+from ... import db
+from ...models import User, DirectAwardProject, AuditEvent, DirectAwardSearch
+from ...utils import (
+    get_json_from_request, get_int_or_400, json_has_required_keys, pagination_links,
+    get_valid_page_or_1, validate_and_return_updater_request, validate_user_can_access_direct_award_project_or_403)
+
+
+@main.route('/direct-award/projects', methods=['GET'])
+def list_projects():
+    user_id = get_int_or_400(request.args, 'user_id')
+    if user_id is None:
+        abort(400, "User ID not supplied")
+
+    page = get_valid_page_or_1()
+
+    projects = DirectAwardProject.query.filter(DirectAwardProject.users.any(id=user_id))
+    projects = projects.order_by(desc(DirectAwardProject.created_at), desc(DirectAwardProject.id))
+
+    projects = projects.paginate(
+        page=page,
+        per_page=current_app.config['DM_API_PROJECTS_PAGE_SIZE'],
+    )
+
+    return jsonify(
+        projects=[project.serialize() for project in projects.items],
+        meta={
+            "total": projects.total,
+        },
+        links=pagination_links(
+            projects,
+            '.list_projects',
+            request.args
+        ),
+    )
+
+
+@main.route('/direct-award/projects', methods=['POST'])
+def create_project():
+    updater_json = validate_and_return_updater_request()
+
+    json_payload = get_json_from_request()
+    json_has_required_keys(json_payload, ['project'])
+
+    project_json = json_payload['project']
+    json_has_required_keys(project_json, ['name', 'user_id'])
+
+    user = User.query.get(project_json.pop('user_id'))
+
+    if user is None:
+        abort(400, "User ID not supplied")
+
+    project = DirectAwardProject(name=project_json['name'], users=[user])
+    db.session.add(project)
+
+    try:
+        db.session.flush()
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, e.orig)
+
+    audit = AuditEvent(
+        audit_type=AuditTypes.create_project,
+        user=updater_json['updated_by'],
+        data={
+            'project_id': project.id,
+            'project_json': project_json,
+        },
+        db_object=project,
+    )
+
+    db.session.add(audit)
+    db.session.commit()
+
+    return jsonify(project=project.serialize()), 201
+
+
+@main.route('/direct-award/projects/<int:project_id>', methods=['GET'])
+def get_project(project_id):
+    user_id = get_int_or_400(request.args, 'user_id')
+    if user_id is None:
+        abort(400, "User ID not supplied")
```

Comment:
Ok, I'll remove the user_id checks. I suppose I thought it added some modicum of extra precaution against displaying data to the wrong user (in case something went weird in the frontend), but maybe in reality that's unlikely to be an issue or one that we don't want to account for.